includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -1165,7 +1171,10 @@ private function get_payment_method_data_from_intent( $intent ) {
 				$payment_method_type    = ! empty( $payment_method_details ) ? $payment_method_details['type'] : '';
 			}
 		} elseif ( 'setup_intent' === $intent->object ) {
-			if ( ! empty( $intent->payment_method ) ) {
+			if ( ! empty( $intent->latest_attempt ) ) {
+				$payment_method_details = (array) $intent->latest_attempt->payment_method_details;
+				$payment_method_type    = ! empty( $payment_method_details ) ? $payment_method_details['type'] : '';
```

Comment:
Looks like maybe we should pall back to using $intent->payment_method if $intent->latest_attempt->payment_method_details is empty. Like:
```suggestion
			if ( ! empty( $intent->latest_attempt ) && ! empty( $intent->latest_attempt->payment_method_details ) ) {
				$payment_method_details = (array) $intent->latest_attempt->payment_method_details;
				$payment_method_type    = $payment_method_details['type'];
```

Reply:
Nice catch, I completely agree.